lua/lspconfig/util.lua

```diff
@@ -294,7 +243,52 @@ function M.strip_archive_subpath(path)
   return path
 end
 
---- Functions that can be removed once minimum required neovim version is high enough
+--- Public functions that can be deprecated once minimum required neovim version is high enough
+
+local function is_fs_root(path)
+  if iswin then
+    return path:match '^%a:$'
+  else
+    return path == '/'
+  end
```

Comment:
```suggestion
  return iswin and path:match '^%a:$' or path == '/'
```

Reply:
I think this is dangerous, if `iswin` is true and `path:match` returns `nil` then it will incorrectly set `path` to `/`. Example:

```lua
print(true and nil or 'oof')
```

prints `oof` even though the intention is to return `nil`

Reply:
Oh i forget there nil trap yes